crates/router/src/types/storage/revenue_recovery.rs

```diff
@@ -40,21 +46,41 @@ impl RevenueRecoveryPaymentData {
             }
             enums::RevenueRecoveryAlgorithmType::Cascading => {
                 revenue_recovery::get_schedule_time_to_retry_mit_payments(
-                    db,
+                    state.store.as_ref(),
                     merchant_id,
                     retry_count,
                 )
                 .await
             }
             enums::RevenueRecoveryAlgorithmType::Smart => {
-                // TODO: Integrate the smart retry call to return back a schedule time
-                None
+                revenue_recovery::get_schedule_time_for_smart_retry(
+                    state,
+                    payment_attempt,
+                    payment_intent,
+                    retry_count,
+                )
+                .await
             }
         }
     }
 }
+
 #[derive(Debug, serde::Deserialize, Clone, Default)]
 pub struct RevenueRecoverySettings {
     pub monitoring_threshold_in_seconds: i64,
     pub retry_algorithm_type: enums::RevenueRecoveryAlgorithmType,
+    pub recovery_timestamp: RecoveryTimestamp,
+}
+
+#[derive(Debug, serde::Deserialize, Clone)]
+pub struct RecoveryTimestamp {
+    pub initial_timestamp_in_hours: i64,
```

Comment:
Why do we need an `i64`, can't we use something like `u16` or `u32`? This is anyway a duration, we don't need these to be negative.